src/main/java/org/springframework/data/repository/config/RepositoryConfigurationDelegate.java

```diff
@@ -133,7 +128,7 @@ private static Environment defaultEnvironment(Environment environment, ResourceL
 			}
 
 			AbstractBeanDefinition beanDefinition = definitionBuilder.getBeanDefinition();
-			String beanName = beanNameGenerator.generateBeanName(beanDefinition, registry);
+			String beanName = configurationSource.getBeanNameGenerator().generateBeanName(beanDefinition);
```

Comment:
Looks like the pattern "get generator to then generate a name" is a very common pattern (seen above as well). Does it maybe make sense to — instead of exposing the generator — introduce a shortcut method on the configuration source to generate a bean name for a definition?